### DISCUSSION TITLE
BPP implementation: when is order tracking mandatory versus optional?

### DISCUSSION THREAD

**Discussion by sudhirrg** [beckn/mobility#42] [Q&A]
If tracking:false is sent as a part of order, does that mean (a) our application does not track the order and (b) does it imply that tracking api need not be implemented?

**Comment by ravi-prakash-v** [comment]
All APIs must be implemented by BPPs. 

The `fulfillment.tracking` is a boolean property. This field can be set to `false` if the BPP does not support real-time tracking for a particular order, in which case calling the `track` endpoint of the BAP will return a "NOT-SUPPORTED" error.

The BAP expects a tracking link from the BPP in the `on_track` in case it receives `tracking:true`.

The BAP can always know if live tracking is not supported by calling the track API and getting an error. But it can avoid calling the BPP if it sees this field set to false during the `on_confirm`, `on_init`, `on_select`, or `on_search` APIs.
